support-files/mariadb-socket-convert

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+#
+# Used to generate a mariadb.socket file based on the curent mysql/maridb settings
+#
+# This is to assist distro maintainers in migrating to systemd socket activation from
+# a user system with networking settings somewhere in the my.cnf files.
+#
+# Redirect output to user directory like /etc/systemd/system/mariadb.socket.d/migrated-from-my.cnf-settings.conf
+
+
+print_ports()
+{
+if [ -n "${port}" ]; then
+  echo "ListenStream=${bind}${port}"
+fi
+if [ -n "${extraport}" ]; then
+  echo "ListenStream=${bind}${extraport}"
+fi
+}
+
+bindaddress=
+port=
+extraport=
+backlog=
+maxconnections=
+skip_networking=
+socket=
+
+coproc mysqldefaults { /usr/bin/my_print_defaults --mysqld ; }
+while read var; do
+  key=${var%%=*}
+  val=${var#*=}
+  case "${key}" in
+    --bind-address)
+      bindaddress=$val
+      ;;
+    --bind_address)
+      bindaddress=$val
+      ;;
+    --port)
+      port=$val
+      ;;
+    --extra-port)
+      extraport=$val
+      ;;
+    --extra_port)
+      extraport=$val
+      ;;
+    --backlog)
+      backlog=$val
+      ;;
+    --max-connections)
+      maxconnections=$val
+      ;;
+    --max_connections)
+      maxconnections=$val
+      ;;
+    --skip-networking)
+      if [ -z "${val}" ]; then
+        skipnetworking=1
+      else
+        skipnetworking=$val
+      fi
+      ;;
+    --skip_networking)
+      if [ -z "${val}" ]; then
+        skipnetworking=1
+      else
+        skipnetworking=$val
+      fi
+      ;;
+    --socket)
+      socket=$val
+      ;;
+  esac
+done < /dev/fd/${mysqldefaults[0]}
+
+echo '[Socket]'
+
+if [ -n "${backlog}" ]; then
+  echo "Backlog=${backlog}"
+elif [ -n "${maxconnections}" ]; then
+  if [ $maxconnections -gt 150 ]; then
+    echo "Backlog=150"
+  else
+    echo "Backlog=${maxconnections}"
+  fi
+fi
+
+if [ -n "${port}" -o \
+     -n "${extraport}" -o \
+    "${skipnetworking}" = '0' -o \
+    -n "${bindaddress}" -o \
+    -n "${socket}" ]; then
+  # we've set unix or inet sockets to non default
+  echo "# reset the ListenStream list to empty"
+  echo "ListenStream="
+  echo
+  # if port isn't set it really needs to be to
+  # fill out the details
+  [ -z "${port}" ] && port=3306
```

Comment:
It doesn't work in all cases, because mysqld can be compiled with
a non-standard port and my_print_defaults will not show it.

I suggest to ignore this corner case in this script :)